### ISSUE TITLE
Get us off of submodules, into virtual env

### ISSUE THREAD

**Issue by bslatkin** [bslatkin/dpxdt#131]
Things to do:

- [x] Move submodules to pip
- [x] Fix flask_mail
- [x] Fix flask_cache
- [x] Update .travis.yml
- [x] Make setup.py install secondary binaries for dpxdt_worker and dpxdt_server
- [x] Rewrite deployment instructions
- [x] Merge dpxdt local instructions into main readme
- [x] Remove HTML from readme

Current contents of .gitmodules is:

```
[submodule "dependencies/flask"]
	path = dependencies/flask
	url = https://github.com/mitsuhiko/flask
[submodule "dependencies/werkzeug"]
	path = dependencies/werkzeug
	url = https://github.com/mitsuhiko/werkzeug.git
[submodule "dependencies/jinja2"]
	path = dependencies/jinja2
	url = https://github.com/mitsuhiko/jinja2.git
[submodule "dependencies/itsdangerous"]
	path = dependencies/itsdangerous
	url = https://github.com/mitsuhiko/itsdangerous
[submodule "dependencies/flask-wtf"]
	path = dependencies/flask-wtf
	url = https://github.com/ajford/flask-wtf.git
[submodule "dependencies/flask-sqlalchemy"]
	path = dependencies/flask-sqlalchemy
	url = https://github.com/mitsuhiko/flask-sqlalchemy
[submodule "dependencies/sqlalchemy"]
	path = dependencies/sqlalchemy
	url = https://github.com/zzzeek/sqlalchemy
[submodule "dependencies/flask-login"]
	path = dependencies/flask-login
	url = https://github.com/maxcountryman/flask-login
[submodule "dependencies/flask-cache"]
	path = dependencies/flask-cache
	url = https://github.com/bslatkin/flask-cache.git
[submodule "dependencies/blinker"]
	path = dependencies/blinker
	url = https://github.com/jek/blinker.git
	ignore = dirty
[submodule "dependencies/gae_mini_profiler"]
	path = dependencies/gae_mini_profiler
	url = https://github.com/kamens/gae_mini_profiler
[submodule "dependencies/flask-mail"]
	path = dependencies/flask-mail
	url = https://github.com/bslatkin/flask-mail.git
[submodule "dependencies/alembic"]
	path = dependencies/alembic
	url = https://bitbucket.org/zzzeek/alembic.git
[submodule "dependencies/mako"]
	path = dependencies/mako
	url = https://github.com/zzzeek/mako
```

**Comment by bslatkin** [comment]
Things to do:

- Move submodules to pip
- Fix flask_mail
- Fix flask_cache
- Update .travis.yml
- Make setup.py install secondary binaries for dpxdt_worker and dpxdt_server
- Rewrite deployment instructions
- Merge dpxdt local instructions into main readme
- Remove HTML from readme

**Comment by danvk** [comment]
I [checkmark-ified][1] this issue -- it's a pretty slick feature of the GitHub issue tracker. You'll see "2/8 tasks completed" on the issues page.

I'm happy to do "Merge dpxdt local instructions into main readme".

[1]: https://github.com/blog/1375%0A-task-lists-in-gfm-issues-pulls-comments

**Comment by bslatkin** [comment]
@danvk Cool! Thanks. If you could do the readme part that'd be great. I've got a branch working on the rest.

**Comment by bslatkin** [comment]
Most of the way there in my branch: https://github.com/bslatkin/dpxdt/tree/virtual_env

Testing a few more things and then will merge it in. Deployment is now much, much easier. The number of dependent packages you need to pull in for the pip package is much higher though. Is that alright with you @danvk? The alternative is to split it into two packages, one for local, one for the server.

**Comment by danvk** [comment]
It feels a little odd for local dpxdt to depend on Flask & AppEngine—do those wind up pulling in a lot of packages transitively? It may not be a big deal in practice.

**Comment by bslatkin** [comment]
@danvk Here's the full list:

https://github.com/bslatkin/dpxdt/blob/virtual_env/requirements.txt

No App Engine dependency, just Flask and its various bits. This may make it easier to launch local UI for showing diffs though?

**Comment by danvk** [comment]
I see, `GoogleAppEngineCloudStorageClient` is a separate thing? Let's give it a try keeping just one package, we can always split if it's a problem.

For launching a local UI to view diffs, I was imagining adding [webdiff][1] as a dependency.

[1]: https://github.com/danvk/webdiff/

**Comment by bslatkin** [comment]
Sorry yeah I got confused :/ We only need that if you deploy to appengine, using this file:

https://github.com/bslatkin/dpxdt/blob/master/deployment/appengine/hooks.py

I can drop that dependency from requirements.txt and put it in the Dockerfile.

Bringing in webdiff sounds good to me.

**Comment by bslatkin** [comment]
Fixed by #145
